cmd/bundle/validate.go

```diff
@@ -140,6 +141,7 @@ func newValidateCommand() *cobra.Command {
 		}
 
 		diags = diags.Extend(bundle.Apply(ctx, b, phases.Initialize()))
+		diags = diags.Extend(bundle.Apply(ctx, b, validate.Validate()))
```

Comment:
Shouldn't these also be applied during deploy?

Reply:
We shouldn't for the set of validation checks that perform I/O.

We can choose to run some of them at deploy time later if 1) they don't add latency, 2) the deploy would fail anyway if the validation fails.

Reply:
Point taken about performance! Alternatively, we group all validations that affect performance together? Rather than have one folder simply called `/validate/` we could have a separate folder like `/slow_validations/` or something. Then other validations like `JobClusterKeyDefined` and most future validations are still applied.

It's important that we include these warnings in the `deploy` path too since that is the common CUJ for CLI users right now.

Reply:
Yeah, indeed. Would fit well in a change of what we display on deploy.